prov/util/src/util_coll.c

```diff
@@ -226,13 +203,89 @@ static inline uint64_t util_coll_form_tag(uint32_t coll_id, uint32_t rank)
 
 static inline uint32_t util_coll_get_next_id(struct util_coll_mc *coll_mc)
 {
-	uint32_t cid = coll_mc->cid;
+	uint32_t cid = coll_mc->group_id;
 	return cid << 16 | coll_mc->seq++;
 }
 
-static int util_coll_sched_send(struct util_coll_mc *coll_mc, uint32_t dest,
+static inline int util_coll_op_init(struct util_coll_operation **coll_op,
```

Comment:
I would rename this to coll_op_create, versus init.  In general, the 'init' calls in OFI are void functions and only setup the structures.